File: add_two_numbers.py
class Solution(object):
    def addTwoNumbers(self, l1, l2):
      
        p1 = l1
        p2 = l2
        
        currentCarry = 0
        
        head = cur = ListNode(0)
        
        while p1 or p2 or currentCarry:
            
            currentVal = currentCarry
            
            currentVal += 0 if p1 is None else p1.val
            currentVal += 0 if p2 is None else p2.val
            
            if currentVal >= 10:
                currentVal -= 10
                currentCarry = 1
            
            else:
                currentCarry = 0
                
            cur.next = ListNode(currentVal)
            cur = cur.next
                
            if p1 is None and p2 is None:
                break
            elif p1 is None:
                p2 = p2.next
            elif p2 is None:
                p1 = p1.next
            else:
                p1 = p1.next
                p2 = p2.next
                
        return head.next
                

File: add_two_numbers.js
var addTwoNumbers = function(l1, l2) {
    let runner1 = l1;
    let runner2 = l2;
    let isOverTen = false;
    let head = new ListNode(null);
    let current = head;
    
    while (runner1 || runner2 || isOverTen) {
        
        let value1 = runner1 ? runner1.val : 0;
        let value2 = runner2 ? runner2.val : 0;
        
        let total = isOverTen ? value1 + value2 + 1 : value1 + value2;
        isOverTen = total > 9 ? true : false;
        total = total % 10;
        
        current.next = new ListNode(total);
        current = current.next;
        
        if (runner1) {
            runner1 = runner1.next
        }
        if (runner2) {
            runner2 = runner2.next;
        }
    }
    
    return head.next;
};
